api/envoy/api/v2/core/health_check_logging.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+package envoy.api.v2.core;
+
+import "envoy/api/v2/core/base.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+
+import "validate/validate.proto";
+import "gogoproto/gogo.proto";
+
+option (gogoproto.equal_all) = true;
+
+// [#protodoc-title: Health check logging]
+// [#proto-status: draft]
+
+message HealthCheckEvent {
+  string host_address = 1 [(validate.rules).string.min_bytes = 1];
+  string cluster_name = 2 [(validate.rules).string.min_bytes = 1];
+
+  oneof event {
+    option (validate.required) = true;
+
+    // Host ejection.
+    HealthCheckEjectUnhealthy eject_unhealthy_event = 3;
+
+    // Host addition.
+    HealthCheckAddHealthy add_healthy_event = 4;
+  }
+}
+
+enum HealthCheckFailureType {
```

Comment:
thanks for adding this. I was also wondering about adding the type of health check that failed (redis, http, tcp).

Reply:
Oh! Right. Sure, will do.